stm32-modules/heater-shaker/firmware/host_comms_task/uart_hardware.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __UART_HARDWARE_H
+#define __UART_HARDWARE_H
+#include "stm32f3xx_hal.h"
+#include "stm32f3xx_hal_uart.h"
+#include "stm32f3xx_hal_uart_ex.h"
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+void UART_Init(UART_HandleTypeDef *huart);
+void UART_DeInit(UART_HandleTypeDef *huart);
+void USART2_IRQHandler(void);
+
+#define UART_BUFFER_MAX_SIZE 64U
+#define UART_BUFFER_MIN_SIZE 16U
```

Comment:
Is this the max number of bytes that one UART callback will contain, or does it come from something else? A comment about the reasoning for this number would be helpful.

Reply:
The max value is an arbitrary value that is significantly larger than any gcode input command. If the remaining buffer size is less than the min value, we may not have enough storage for a subsequent gcode input.